File: PyPI/src/spreen_readme/application.py
"""Updates the README.md of a repository.

The packaging around it is complete; the update logic itself is still to be
written, so `run` currently reports the target and succeeds without touching
anything."""

import os
import sys
from typing import TextIO


class Application:
    """Updates the README.md of a repository."""

    @classmethod
    def run(cls, path: str = '.', io: TextIO | None = None) -> int:
        return cls(path=path, io=io)._run()

    def __init__(self, path: str = '.', io: TextIO | None = None) -> None:
        self.path = path
        self.io = io if io is not None else sys.stdout

    # private

    def _run(self) -> int:
        print(f'Target README is {self._readme_path()}', file=self.io)
        return 0

    def _readme_path(self) -> str:
        return os.path.abspath(os.path.join(self.path, 'README.md'))

File: RubyGem/lib/spreen_readme/application.rb
# frozen_string_literal: true
# rbs_inline: enabled

module SpreenReadme
  # Updates the README.md of a repository.
  #
  # The packaging around it is complete; the update logic itself is still to
  # be written, so `run` currently reports the target and succeeds without
  # touching anything.
  class Application
    # @rbs path: String
    # @rbs io: IO
    # @rbs return: Integer
    def self.run(path: '.', io: $stdout)
      new(path:, io:).run
    end

    # @rbs path: String
    # @rbs io: IO
    # @rbs return: void
    def initialize(path: '.', io: $stdout)
      @path = path
      @io   = io
    end

    # @rbs return: Integer
    def run
      io.puts "Target README is #{readme_path}"
      0
    end

    private

    attr_reader :path #: String
    attr_reader :io #: IO

    # @rbs return: String
    def readme_path
      File.absolute_path(File.join(path, 'README.md'))
    end
  end
end
